back/app/services/idea_custom_fields_service.rb

```diff
@@ -80,6 +80,26 @@ def remove_ignored_update_params(field_params)
     field_params.except(:code, :input_type)
   end
 
+  def check_form_structure(fields, errors)
+    participation_method = @custom_form.participation_context.participation_method
+    can_have_type = participation_method == 'ideation' ? 'section' : 'page'
```

Comment:
In general, we should try to avoid conditionals on the participation method. This is done by moving the logic to the participation methods (in the lib folder).

Reply:
I think we should rename this class to `InputCustomFieldsService` at some point, because it's not about just ideas; it's about inputs in general.

Reply:
@sebastienhoorens I've changed this logic as suggested, but not sure now is the right time to tackle renaming this class - if we go down that route then we could in all probability end up wanting to rename the controller, model etc too!